nomos-da/network/subnetworks-assignations/src/lib.rs

```diff
@@ -11,6 +14,11 @@ pub trait MembershipHandler {
     /// Members Id type
     type Id;
 
+    /// Creates a new instance of the membership handler with the given members
+    /// and address book
+    #[must_use]
+    fn new_with(&self, members: Vec<PeerId>, addressbook: HashMap<PeerId, Multiaddr>) -> Self;
```

Comment:
@danielSanchezQ I am answering here as comments are broken in the main PR.

The reason for this factory method is that v1 and v2 assignation (membership handler) is created generically. The initial members and addressbook are read from the config. The config also includes configuration values for v1 and v2 such as dispersal factor etc.. This method just enables the caller to create a new one with the same configuration but difdferent addressbook and members.

I did not want to create a separate trait for this call only, as:
1. It will introduce a new type and might bloat types a bit
2. The users can not really abuse this as they can not modify anything